.github/workflows/nuget-publish.yml

```diff
@@ -0,0 +1,70 @@
+name: NuGet Publish
+
+on:
+  push:
+    branches: [ master ]
+
+env:
+  NUGET_DIR: ${{ github.workspace }}/nuget
+  NUGET_ARTIFACT: nuget
+
+jobs:
+  testing:
+    if: contains(github.event.head_commit.message, 'chore(release)') 
+    uses: ./.github/workflows/test.yml
+
+  pack: 
+    if: contains(github.event.head_commit.message, 'chore(release)') 
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v3
+
+      - run: sed -E 's:RepositoryCommit>.+<\/:RepositoryCommit>${{ github.sha }}<\/:g' ButterCMS/ButterCMS.csproj > tmp && mv tmp ButterCMS/ButterCMS.csproj
+
+      - run: dotnet pack --configuration Release --output ${{ env.NUGET_DIR }}
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.NUGET_ARTIFACT }}
+          path: ${{ env.NUGET_DIR }}
+          if-no-files-found: error
+          retention-days: 1
+
+  validate-nuget:
+    runs-on: ubuntu-latest
+    needs: pack
+    steps:
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v3
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: ${{ env.NUGET_ARTIFACT }}
+          path: ${{ env.NUGET_DIR }}
+
+      - name: Install nuget validator
+        run: dotnet tool update Meziantou.Framework.NuGetPackageValidation.Tool --global
+
+      - name: Validate nuget package
+        run: meziantou.validate-nuget-package $(ls ${{ env.NUGET_DIR }}/*.nupkg | head -n 1) --excluded-rules Symbols
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: [validate-nuget, testing]
+    steps:
+      - uses: actions/download-artifact@v3
+        with:
+          name: ${{ env.NUGET_ARTIFACT }}
+          path: ${{ env.NUGET_DIR }}
+
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v3
+
+      - name: Publish nuget package
+        run: dotnet nuget push $(ls ${{ env.NUGET_DIR }}/*.nupkg | head -n 1) --api-key "${{ secrets.NUGET_APIKEY }}" --source https://api.nuget.org/v3/index.json
```

Comment:
shellcheck:
```

In a.sh line 1:
dotnet nuget push $(ls ${{ env.NUGET_DIR }}/*.nupkg | head -n 1) --api-key "${{ secrets.NUGET_APIKEY }}" --source https://api.nuget.org/v3/index.json
                  ^-- SC2046 (warning): Quote this to prevent word splitting.
                    ^-----------------------------^ SC2012 (info): Use find instead of ls to better handle non-alphanumeric filenames.
                       ^-----------------^ SC2086 (info): Double quote to prevent globbing and word splitting.
                         ^--------------^ SC2296 (error): Parameter expansions can't start with {. Double check syntax.
                                          ^-- SC1083 (warning): This } is literal. Check expression (missing ;/\n?) or quote it.
                                                                              ^---------------------^ SC2296 (error): Parameter expansions can't start with {. Double check syntax.

Did you mean: 
dotnet nuget push $(ls "${{ env.NUGET_DIR }"}/*.nupkg | head -n 1) --api-key "${{ secrets.NUGET_APIKEY }}" --source https://api.nuget.org/v3/index.json

For more information:
  https://www.shellcheck.net/wiki/SC2296 -- Parameter expansions can't start ...
  https://www.shellcheck.net/wiki/SC1083 -- This } is literal. Check expressi...
  https://www.shellcheck.net/wiki/SC2046 -- Quote this to prevent word splitt...
```

Reply:
Resolved in https://github.com/ButterCMS/buttercms-csharp/pull/40/commits/f3d2d5e3f00c7bfcdd3405cd42a7cfb56b556a32

Reply:
`dotnet nuget push` seem to support pushing multiple files at once:
> Push all _.nupkg_ files in the current directory to the default push source specified in the NuGet config file:
> ```
> dotnet nuget push "*.nupkg"
> ```
> —https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-nuget-push

Therefore, that for-loop may be redundant.

Reply:
One last thing 😅 
The docs specify that:
```
dotnet nuget push "*.nupkg"
```
> Note
> The enclosing quotes are required for shells such as bash that perform file globbing. For more information, see [NuGet/Home#4393](https://github.com/NuGet/Home/issues/4393#issuecomment-667618120).

So I think we need to do:
```
dotnet nuget push "${{ env.NUGET_DIR }}/*.nupkg" …
```

(I didn't test that, tho)

Reply:
@Toreno96 tested and resolved in https://github.com/ButterCMS/buttercms-csharp/pull/40/commits/23ec3acc174e05d1bf6516ee3a68016387ab7211